qutip/core/coefficient.py

```diff
@@ -47,8 +47,24 @@ class StringParsingWarning(Warning):
     pass
 
 
+def _return(base, **kwargs):
+    return base
+
+
+# The `coefficient` function is dispatcher for the type of the `base` to the
+# function that create the `Coefficient` object. `coefficient_builders` store
+# the map `type -> function(base, **kw)`. Optional module can add their
+# `Coefficient` specialization here.
+coefficient_builders = {
+    Coefficient: _return,
+    np.ndarray: InterCoefficient,
+    scipy.interpolate.PPoly: InterCoefficient.from_PPoly,
+    scipy.interpolate.BSpline: InterCoefficient.from_Bspline,
+}
+
+
 def coefficient(base, *, tlist=None, args={}, args_ctypes={},
```

Comment:
The documentation for this function is great! although I still miss the Parameters section. For instance, it took me some digging to understand what kwargs are meant to be used for. Is it mainly used to supprt the builders from jax?